### ISSUE TITLE
Compilation fails on Archlinux 

### ISSUE THREAD

**Issue by jonas-w** [drolbr/Overpass-API#703]
I ran:
```bash
./configure --disable-dependency-tracking --enable-lz4
make -j4
```
And this error occurs on '0.7.61.6':
```
template_db/file_tools.cc: In member function ‘uint32 Blocking_Client_Socket::get_command()’:
template_db/file_tools.cc:82:28: error: ‘uint8_t’ was not declared in this scope
   82 |       socket_descriptor, ((uint8_t*)&buffer[0])+bytes_in_buffer, buffer.size()*sizeof(uint32) - bytes_in_buffer, 0);
      |                            ^~~~~~~
template_db/file_tools.cc:38:1: note: ‘uint8_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
   37 | #include <iostream>
  +++ |+#include <cstdint>
   38 | #include <vector>
template_db/file_tools.cc:82:36: error: expected primary-expression before ‘)’ token
   82 |       socket_descriptor, ((uint8_t*)&buffer[0])+bytes_in_buffer, buffer.size()*sizeof(uint32) - bytes_in_buffer, 0);
      |                                    ^
```

Same thing works on Debian. Is there a way to compile it on Archlinux?

**Comment by mmd-osm** [comment]
Can you try to reach out to your Archlinux community and discuss it with them in a first step? They might be in a better position to point you to Archlinux specific issues.

**Comment by drolbr** [comment]
> Can you try to reach out to your Archlinux community and discuss it with
> them in a first step? They might be in a better position to point you to
> Archlinux specific issues.

No, Jonas is correctly reporting here. There is a missing
````
#include <cstdint>
````
in apparently more places than it looked like from the Kali problem.

I do not know why neither GCC nor Clang detect this even in pedantic
mode on other platforms, but it makes it rater tricky to sport all places.

A workaround would be to add
````
#include <cstdint>
````
in all files where the compiler complains, but that could be as well
subject of a patch release.

**Comment by mmd-osm** [comment]
I thought this might be something ArchLinux specific, like in https://github.com/AirenSoft/OvenMediaEngine/issues/1271 ... The issue suggests that the missing "std::" namespace for uint8_t is causing the issue.

Maybe worth noting: They have resolved the issue by including stdint.h since they prefer using uint8_t over std::uint8_t.

**Comment by drolbr** [comment]
On 22.08.23 00:24, Jonas Wunderlich wrote:
> I ran:
>
> |./configure --disable-dependency-tracking --enable-lz4 make -j4 |
>
> And this error occurs on '0.7.61.6':
[..]

Thank you for reporting this. It should be fixed in 0.7.61.7

**Comment by mmd-osm** [comment]
I tested this on an ArchLinux docker image now. Like the compiler error message above states, the only missing thing  was an `#include <cstdint>` in template_db/file_tools.cc line 38.

template_db/file_tools.cc:38:1: note: ‘uint8_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?

**Comment by drolbr** [comment]
I'm sorry for putting it in the wrong branch. Should be fixed in 0.7.61.8
